kernel/kernel-api/src/main/java/io/delta/kernel/internal/deletionvectors/RoaringBitmapArray.java

```diff
@@ -258,17 +259,39 @@ private void extendBitmaps(int newLength) {
         newBitmaps, // dest
         0, // dest start pos
         bitmaps.length); // number of entries to copy
-    for (int i = 0; i < bitmaps.length; i++) {
+    for (int i = bitmaps.length; i < newBitmaps.length; i++) {
       newBitmaps[i] = new RoaringBitmap();
     }
     bitmaps = newBitmaps;
   }
 
   public static RoaringBitmapArray create(long... values) {
     RoaringBitmapArray bitmap = new RoaringBitmapArray();
+    bitmap.bitmaps = new RoaringBitmap[0];
     for (long value : values) {
       bitmap.add(value);
     }
     return bitmap;
   }
+
+  public long[] toArray() {
```

Comment:
Any particular reason for adding this?

I think I would prefer to either
(1) not add this for now since we don't use it
(2) if we are adding it, it needs more tests

I would lean toward (1) since it's not used but if you have a good reason for adding it lmk

Reply:
One reason I think is that with the current methods available is not possible to assert that the contents of the data structure are exactly the expected. 
The unit test I added  tests the `contains` method for some particular values that are expected or not expected to be there, but it cannot completely discard that an unwanted value is present. With this method it can be asserted that the wanted values are there, and only them. 

Another one, is somehow related. Without this method any implementation of the scanning of files is constrained to use a filter approach to read the data, and call the `contains` for each single row. Depending on the number and the relation between total rows and deleted rows, and how the file is processed, I think it can be interesting to also allow the approach of reading the whole rows and then discarding the ones deleted.

Reply:
Okay I'm still okay with either (1) or (2). But if we are going with (2) can you please add full test coverage?

i.e. cover things like
- varios number of underlying bitmaps
- empty case
- etc